apps/test/unit/templates/sectionsRefresh/CoteacherSettingsTest.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import {mount, shallow} from 'enzyme';
+import {expect} from '../../../util/reconfiguredChai';
+import CoteacherSettings from '@cdo/apps/templates/sectionsRefresh/CoteacherSettings';
+import sinon from 'sinon';
+
+const testSectionInstructors = [
+  {instructorName: 'Allosaurus', instructorEmail: 'allosaurus@code.org'},
+  {instructorName: 'Brachiosaurus', instructorEmail: 'brachiosaurus@code.org'},
+];
+
+describe('CoteacherSettings', () => {
+  it('renders count of existing coteachers', () => {
+    const wrapper = mount(
+      <CoteacherSettings
+        sectionInstructors={testSectionInstructors}
+        addCoteacher={() => {}}
+        coteachersToAdd={[]}
+      />
+    );
+    expect(wrapper.find('Figcaption')).to.have.lengthOf(1);
+    expect(wrapper.find('Figcaption').text()).to.include(
+      '1/5 co-teachers added'
+    );
+  });
+  it('renders count of existing coteachers and added', () => {
+    const wrapper = mount(
+      <CoteacherSettings
+        sectionInstructors={testSectionInstructors}
+        addCoteacher={() => {}}
+        coteachersToAdd={['coelophysis@code.org']}
+      />
+    );
+    expect(wrapper.find('Figcaption')).to.have.lengthOf(1);
+    expect(wrapper.find('Figcaption').text()).to.include(
+      '2/5 co-teachers added'
+    );
+  });
+  it('disables add button when max coteachers reached', () => {
+    const wrapper = shallow(
+      <CoteacherSettings
+        sectionInstructors={testSectionInstructors}
+        addCoteacher={() => {}}
+        coteachersToAdd={[
+          'coelophysis@code.org',
+          'diplodocus@code.org',
+          'eoraptor@code.org',
+          'fylax@code.org',
+        ]}
+      />
+    );
+    expect(wrapper.find('Button').first().props().disabled).to.be.true;
+  });
+  it('shows error when adding invalid email', () => {
```

Comment:
I am not sure if you could move to `shallow` here using the same logic... maybe test it out?